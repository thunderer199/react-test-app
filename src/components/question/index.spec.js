import React from "react";

import renderer from 'react-test-renderer'

import Question from "./index";
import {shallow} from "enzyme";

describe('Question', () => {
    it('should be rendered', () => {
        const div = document.createElement('div');
        const payload = {name: 'test', answers: []};
        shallow(<Question question={payload}/>, div);
    });

    it('should render payload right without answers', () => {
        const payload = {name: 'test'};
        const snapshot = renderer.create(
            <Question question={payload}/>
        ).toJSON();

        expect(snapshot).toMatchSnapshot();
    });

    it('should render payload right with answers', () => {
        const payload = {name: 'test', answers: [{id: 1, value: 'some answer'}, {id: 2, value: 'some answer'}]};
        const snapshot = renderer.create(
            <Question question={payload}/>
        ).toJSON();

        expect(snapshot).toMatchSnapshot();
    });

    it('should render payload right with answers', () => {
        const fn = jest.fn();
        const payload = {id: 1, name: 'test', answers: [{id: 1, value: 'some answer'}, {id: 2, value: 'some answer'}]};
        const component = shallow(<Question question={payload} dispatch={fn}/>);
        component.find('li').at(1).simulate('click');

        expect(fn).toHaveBeenCalledWith({question: 1, answer: 2});
    });
});