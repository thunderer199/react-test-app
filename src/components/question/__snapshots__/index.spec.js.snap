// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should render payload right with answers 1`] = `
<div>
  <h4>
    test
  </h4>
  <ul>
    <li
      className="answer-element"
      onClick={[Function]}
    >
      some answer
    </li>
    <li
      className="answer-element"
      onClick={[Function]}
    >
      some answer
    </li>
  </ul>
</div>
`;

exports[`Question should render payload right without answers 1`] = `
<div>
  <h4>
    test
  </h4>
  <ul />
</div>
`;
